client/my-sites/controller.js

```diff
@@ -109,6 +111,50 @@ function renderNoVisibleSites( context ) {
 	);
 }
 
+function isPathAllowedForDomainOnlySite( pathname, site ) {
+	if ( ! get( site, 'options.is_domain_only', false ) ) {
```

Comment:
That would be nice. It's quite easy to pass state as param. It's available in the parent function that calls this one.